frontend/src/app/modules/matches/matches-computations.js

```diff
@@ -68,3 +77,83 @@ export const computeWinRatio = (playerId, playerMatches) => {
   const wonMatchesCount = playerMatches.filter(match => didPlayerWin(playerId, match)).length
   return (playerMatches.length > 0) ? (wonMatchesCount / playerMatches.length) : 0
 }
+
+export const computeWins = (playerId, playerMatches) =>
+  playerMatches.filter(match => didPlayerWin(playerId, match)).length
+
+const findWithComparator = (comparator, obj, field, baseScore) =>
+  Object.keys(obj).reduce((acc, key) =>
+    comparator(acc.score, obj[key][field])
+      ? acc
+      : { value: obj[key].value, score: obj[key][field] }
+  , { score: baseScore })
+
+export const findMin = (obj, field = 'score') =>
+  findWithComparator((e1, e2) => e1 <= e2, obj, field, Infinity)
+
+export const findMax = (obj, field = 'score') =>
+  findWithComparator((e1, e2) => e1 >= e2, obj, field, -Infinity)
+
+export const computeDays = matchChanges => {
+  const daysMap = {}
+
+  for (const match of matchChanges) {
+    const day = match.date.toLocaleDateString()
+    const score = Number(match.ratingChangeString)
+    if (!Object.hasOwnProperty.call(daysMap, day)) {
+      daysMap[day] = {
+        value: day,
+        score,
+      }
+    } else {
+      daysMap[day].score += score
+    }
+  }
+  return matchChanges.length
+    ? daysMap
+    : {
+      default: {
+        value: '',
+        score: 0,
+      },
+    }
+}
+
+export const computeTeammates = (playerId, playerMatches) =>
+  computePlayers(playerId, playerMatches, getTeamMates) || {}
+
+export const computeOpponents = (playerId, playerMatches) =>
+  computePlayers(playerId, playerMatches, getOpponents) || {}
+
+const computePlayers = (playerId, playerMatches, playersProvider) => {
+  const playersMap = {}
+
+  for (const match of playerMatches) {
+    const players = playersProvider(playerId, match)
+    players.map(player =>
```

Comment:
better to use forEach in this case than map as we are not mapping here anything...

Reply:
Definitely a good point, in addition, `map` ideally shouldn't have side effects. `forEach` would be more suitable here

Reply:
Good idea, done [here](https://github.com/salsita/foosball-rating/pull/114/commits/48d97a59bd3178627233fc0798f8d61a47a09c3b#diff-744525510377a0aca19d17cf4dc35116), [here](https://github.com/salsita/foosball-rating/pull/114/commits/48d97a59bd3178627233fc0798f8d61a47a09c3b#diff-e1417bef467b606501c1db87ed6132c1) and below

Reply:
Done [here](https://github.com/salsita/foosball-rating/pull/114/commits/48d97a59bd3178627233fc0798f8d61a47a09c3b#diff-304dca28d92d292252e86b7e83fc1569)